tests/pull-test.sh

```diff
@@ -162,6 +162,23 @@ $OSTREE show main >/dev/null
 echo "ok pull local mirror"
 
 cd ${test_tmpdir}
+# This is more of a known issue; test that we give a clean error right now
+rm otherrepo -rf
+ostree_repo_init otherrepo --mode=archive
+rm checkout-origin-main -rf
+${CMD_PREFIX} ostree --repo=ostree-srv/gnomerepo checkout main checkout-origin-main
+${CMD_PREFIX} ostree --repo=otherrepo commit -b localbranch --tree=dir=checkout-origin-main
+${CMD_PREFIX} ostree --repo=otherrepo remote add --set=gpg-verify=false origin file://$(pwd)/ostree-srv/gnomerepo
+${CMD_PREFIX} ostree --repo=otherrepo pull origin main
+rm mirrorrepo-local -rf
+ostree_repo_init mirrorrepo-local --mode=archive
+if ${CMD_PREFIX} ostree --repo=mirrorrepo-local pull-local otherrepo 2>err.txt; then
+    fatal "pull with mixed refs succeeded?"
+fi
+assert_file_has_content err.txt "error: Invalid ref name origin:main"
+echo "ok pull local mirror error (mixed refs)"
```

Comment:
Nit: both this test and the previous test use as ok strings "ok pull local mirror ..." even though in the former, it's talking about `pull --mirror` from a local repo, and here it's talking about `pull-local` from a local repo. Maybe change this one to just e.g. `ok pull-local error (mixed refs)` ?